edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeOperator.cs

```diff
@@ -861,6 +894,47 @@ VolumeOptions GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> mod
                 : Option.None<(List<V1Volume>, List<V1VolumeMount>, List<V1VolumeMount>)>();
         }
 
+        private async Task GetOrCreatePersistentVolumeClaim(string name, bool readOnly)
+        {
+            var listResult = await this.client.ListNamespacedPersistentVolumeClaimAsync(KubeUtils.K8sNamespace);
+
+            var foundPvc = listResult.Items.SingleOrDefault(item => !string.IsNullOrWhiteSpace(item.Metadata.Name) && item.Metadata.Name == name);
+
+            if (foundPvc != default(V1PersistentVolumeClaim))
+            {
+                // ReadWriteOnce is the Kubernetes moniker for single pod use only
+                if (foundPvc.Spec.AccessModes.Contains("Once"))
+                {
+                    // TODO : should we throw here or should we just let Kube throw when we try to mount it?
+                    throw new AccessViolationException($"PVC with name {name} is configured to only be mounted to one node.");
+                }
+
+                // PVC is fine and we can return
+                return;
+            }
+
+            var persistentVolumeClaimSpec = new V1PersistentVolumeClaimSpec()
+            {
+                AccessModes = new List<string> { readOnly ? "ReadOnlyMany" : "ReadWriteMany" },
+                Resources = new V1ResourceRequirements()
+                {
+                    Requests = new Dictionary<string, ResourceQuantity>() { { "storage", new ResourceQuantity($"{this.persistentVolumeClaimSizeMb}Mi") } }
+                }
+            };
+
+            if (this.persistentVolumeNameIsSc)
+            {
+                persistentVolumeClaimSpec.StorageClassName = this.persistentVolumeName;
+            }
+            else
+            {
+                persistentVolumeClaimSpec.VolumeName = this.persistentVolumeName;
+            }
+
+            // TODO : Check return?
+            await this.client.CreateNamespacedPersistentVolumeClaimAsync(new V1PersistentVolumeClaim(metadata: new V1ObjectMeta(name: name), spec: persistentVolumeClaimSpec), KubeUtils.K8sNamespace);
```

Comment:
Regarding the `TODO` above, what should we be checking here that we aren't?

Reply:
This is something that's lacking in the preview code.  We are just letting exceptions be thrown on creation errors. Fixing this should be part of the move to GA.